functions/prepare-check/index.js

```diff
@@ -0,0 +1,99 @@
+const R = require('ramda')
+
+/**
+ * Write to Table Storage for fast pupil authentication
+ * Reads from a queue
+ * @param context
+ * @param {} prepareCheckMessage
+ */
+module.exports = function (context, prepareCheckMessage) {

```

Comment:
For the queue runner I can see performance implications if there is a singular message being processed at once with the round trip time. Since it's a polled solution it may be worthwhile to obtain batches at a time and process them concurrently if possible, to improve performance.

Reply:
the listener is abstract to the function implementation, we just declare the binding.  We can configure some processing behaviours, but we can explore that once we have the solution wired up and working.